stdlib/public/Concurrency/Actor.cpp

```diff
@@ -935,356 +1034,370 @@ static void traceJobQueue(DefaultActorImpl *actor, Job *first) {
+    // TODO (rokhinip): Don't we need to take a +1 per ref count rules specified?
+    swift_retain(this);
+    job = new ProcessOutOfLineJob(this, priority);
+  }
+  SWIFT_TASK_DEBUG_LOG("Scheduling processing job %p for actor %p at priority %#x", job, this, priority);
+  swift_task_enqueueGlobal(job);
+}
 
-    auto firstPreprocessed = oldState.FirstJob;
-    if (!CurrentState.compare_exchange_weak(oldState, newState,
-                    /*success*/ std::memory_order_release,
-                    /*failure*/ std::memory_order_acquire)) {
-      // Preprocess any new queue items.
-      firstNewJob = preprocessQueue(oldState.FirstJob,
-                                    firstPreprocessed,
-                                    firstNewJob);
-      traceJobQueue(this, firstNewJob);
-
-      // Try again.
-      continue;
-    }
 
-#define ACTOR_STATE_TRANSITION                                                 \
-  do {                                                                         \
-    SWIFT_TASK_DEBUG_LOG("actor %p transitioned from %zx to %zx (%s)\n", this, \
-                         oldState.Flags.getOpaqueValue(),                      \
-                         newState.Flags.getOpaqueValue(), __FUNCTION__);       \
-    concurrency::trace::actor_state_changed(                                   \
-        this, newState.FirstJob.getRawJob(),                                   \
-        newState.FirstJob.needsPreprocessing(),                                \
-        newState.Flags.getOpaqueValue());                                      \
-  } while (0)
-    ACTOR_STATE_TRANSITION;
-
-    // The priority of the remaining work.
-    auto newPriority = newState.Flags.getMaxPriority();
-
-    // This is the actor's owning job; per the ownership rules (see
-    // the comment on DefaultActorImpl), if there are remaining
-    // jobs, we need to balance out our ownership one way or another.
-    // We also, of course, need to ensure that there's a thread that's
-    // actually going to process the actor.
-    if (hasMoreJobs) {
-      // Enqueue a job that will try to take over running with the
-      // new priority.  This also ensures that there's a job at that
-      // priority which will actually take over the actor.
-      scheduleNonOverrideProcessJob(newPriority, hasActiveInlineJob);
+bool DefaultActorImpl::tryLock(bool asDrainer) {
+#if SWIFT_CONCURRENCY_ENABLE_PRIORITY_ESCALATION
+  SWIFT_TASK_DEBUG_LOG("Thread %#x attempting to jump onto %p, as drainer = %d", dispatch_lock_value_for_self(), this, asDrainer);
+  dispatch_thread_override_info_s threadOverrideInfo;
+  threadOverrideInfo = swift_dispatch_thread_get_current_override_qos_floor();
+  qos_class_t overrideFloor = threadOverrideInfo.override_qos_floor;
+
+retry:;
+#else
+  SWIFT_TASK_DEBUG_LOG("Thread attempting to jump onto %p, as drainer = %d", this, asDrainer);
+#endif
+
+  auto oldState = CurrentState.load(std::memory_order_relaxed);
```

Comment:
Oh I think this should be a load acquire.

Reply:
I can't tell if it's okay for this to be relaxed or not since the FirstJob is in the atomic so we should be seeing everything in the same double-wide atomic state...

Reply:
Nothing in here is chasing a pointer out of `oldState` so relaxed seems right.

Reply:
Not locally in this function but once we tryLock as a drainer, we typically go into `drainOne` which looks at the queue pointed at by FirstJob in the state and chases that LL of Jobs.

Reply:
The thing I don't know is if we need an `acquire` since we see the head of the LL in the same atomic state so we're not coordinating with a write to a different memory location.

Reply:
I think you'd need the `acquire` if this returned the status or `FirstJob` that was then used to look into the queue, but it doesn't and `drainOne` has its own acquire load.